src/validation.cpp

```diff
@@ -5033,15 +5035,19 @@ bool ChainstateManager::ActivateSnapshot(
     return true;
 }
 
-static void FlushSnapshotToDisk(CCoinsViewCache& coins_cache, bool snapshot_loaded)
+[[nodiscard]] static bool FlushSnapshotToDisk(CCoinsViewCache& coins_cache, bool snapshot_loaded)
 {
     LOG_TIME_MILLIS_WITH_CATEGORY_MSG_ONCE(
         strprintf("[snapshot] %s coins cache (%.2f MB)",
                   (snapshot_loaded ? "syncing" : "flushing"),
                   coins_cache.DynamicMemoryUsage() / (1000 * 1000)),
         BCLog::LogFlags::ALL);
 
-    snapshot_loaded ? coins_cache.Sync() : coins_cache.Flush();
+    if (snapshot_loaded ? coins_cache.Sync() : coins_cache.Flush()) {
+        return true;
+    } else {
+        return AbortNode("Failed to write to coin database");
```

Comment:
Same question about `AbortNode` vs. `error` though.

Reply:
`AbortNode()` is used elsewhere in the function that this function is called from, although I did change another reference that you pointed out what inconsistent.